src/builder/commands/pip.rs

```diff
@@ -313,31 +314,59 @@ impl BuildStep for Py3Install {
     }
 }
 
-fn version_req(hash: &mut Digest, fname: &Path) -> Result<(), VersionError> {
-    let path = Path::new("/work").join(fname);
-    let err = |e| VersionError::Io(e, path.clone());
-    File::open(&path)
-    .and_then(|f| {
-            let f = BufReader::new(f);
-            for line in f.lines() {
-                let line = try!(line);
-                let chunk = line[..].trim();
-                // Ignore empty lines and comments
-                if chunk.len() == 0 || chunk.starts_with("#") {
-                    continue;
-                }
-                // Should we also ignore the order?
-                hash.item(chunk);
-            }
-            Ok(())
-    }).map_err(err)
+fn parse_req_filename(line: &str) -> Option<&str> {
+    let res = vec!["-r", "--requirement ", "--requirement=",
+                   "-c", "--constraint ", "--constraint="];
+    for prefix in res.iter() {
+        if line.starts_with(prefix) {
+            return Some(line[prefix.len()..].trim());
+        }
+    }
+    return None;
+}
+
+fn version_req(hash: &mut Digest, fname: &Path, used: &mut HashSet<String>) ->
+               Result<(), VersionError> {
+
+    let path = try!(Path::new("/work")
+                       .join(fname)
+                       .canonicalize()
+                       .map_err(|e| VersionError::Io(e, fname.to_path_buf())));
+
+    let name = format!("{:?}", path);
+    if used.contains(&name[..]) {
+        return Ok(())
```

Comment:
for dependencies [yes](http://pip-python3.readthedocs.io/en/latest/reference/pip_install.html#installation-order)